src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -841,8 +841,12 @@ public bool Equals(Guid g)
             // Now compare each of the elements
             return g._a == _a &&
```

Comment:
This may be a dumb question that I could have just looked up, but I wanted to ask here so I was sure...  What part of the GUID structure is the LSB here?  Is it different based on big-endian vs. little-endian?  The reason I ask is that if someone is comparing lots of sequentially assigned GUIDs, it's probably faster to compare LSB first, rather than MSB, right?  Seems like that would be faster on average than the other short-circuit eval and not slower in the non-sequential case.

Reply:
@kellypleahy it depends on what you mean by "sequentially assigned GUIDs". For instance MS SQL Server increment first int field something like this:
`00000000-0000-0000-0000-000000000000`
`00010000-0000-0000-0000-000000000000`

Reply:
Ah, thanks.  It appears that UuidCreateSequential at least seems to update _a when generating sequential guids (and SQL Server just reverses the bytes of _a to big-endian when storing so as to ensure sequential order of the byte array) so in any case, comparing _a first is best by my argument above.  Thanks for indulging me.